xonsh/jobs.py

```diff
@@ -199,6 +199,55 @@ def add_job(info):
         print_one_job(num)
 
 
+def clean_jobs():
+    """Clean up jobs for exiting shell
+
+    In non-interactive mode, kill all jobs.
+
+    In interactive mode, check for suspended or background jobs, print a
+    warning if any exist, and return False. Otherwise, return True.
+    """
+    jobs_clean = True
+    if builtins.__xonsh_env__['XONSH_INTERACTIVE']:
+        _clear_dead_jobs()
+        job_types = set()
+        for job in builtins.__xonsh_all_jobs__.values():
+            if job['bg']:
+                job_types.add('background')
+            elif job['status'] == 'stopped':
+                job_types.add('stopped')
+
+        if job_types:
+            if 'background' in job_types and 'stopped' in job_types:
+                type_str = 'background and stopped'
+            else:
+                type_str = job_types.pop()
+
+            if len(builtins.__xonsh_all_jobs__) > 1:
+                num = 'are'
+                job_str = 'jobs'
+            else:
+                num = 'is a'
+                job_str = 'job'
+
+            print()
+            print('xonsh: there {num} {typ} {job}'.format(num=num,
+                                                          typ=type_str,
+                                                          job=job_str))
+            print('-'*5)
+            jobs([])
+            print('-'*5)
+            resp = input('Enter "exit" or "e" to kill all jobs and exit ')
```

Comment:
Also it would be nice if the user could simply hit whatever key that they were already using to exit.  IPython does this with ^C and ^D.  Users shouldn't need to type an extra word.